src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -480,6 +480,21 @@ func overprovisioned(developerMode bool, limits corev1.ResourceList) []string {
 	}
 }
 
+func (r *StatefulSetResource) pandaproxyEnvVars() []corev1.EnvVar {
+	var envs []corev1.EnvVar
+	listener := r.pandaCluster.PandaproxyAPIExternal()
+	if listener != nil {
+		envs = append(envs, corev1.EnvVar{
+			Name:  "PROXY_EXTERNAL_CONNECTIVITY",
```

Comment:
do we need this? We already add PandaproxyPortExternalName listener to the configmap so can't we use that to figure out that there's external connectivity instead of introducing another env var?

Reply:
No, I was on the fence on this one because we already follow this pattern for the Kafka listener.

I removed the bool and infer it instead.